dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -1057,7 +1061,14 @@ private String getNullTerminatedString(HpccSrcType stype, boolean shouldTrim) th
                     readSize = OPTIMIZED_STRING_READ_AHEAD;
                 }
 
-                this.inputStream.mark(readSize);
+                if ((strByteLen + readSize) >= readLimit)
+                {
+                    readLimit += (readLimit * NULL_SCAN_READ_LIMIT_GROW_MULTIPLIER);
```

Comment:
Is NULL_SCAN_READ_LIMIT_GROW_MULTIPLIER supposed to grow? I see it set to 1, but never updated.

Reply:
I think I might need a better name here. That is supposed to represent a percentage to grow the read limit each growth. In this case I have it defaulting to growing by 100% each growth.